src/commands/config/mod.rs

```diff
@@ -17,10 +17,40 @@ pub fn set_file_mode(file: &PathBuf) {
         .expect("could not set permissions on file");
 }
 
-pub fn global_config(email: String, api_key: String) -> Result<(), failure::Error> {
-    let s = GlobalUser { email, api_key };
+pub fn global_config(api_key: bool) -> Result<(), failure::Error> {
+    let mut user = GlobalUser {
+        email: None,
+        api_key: None,
+        api_token: None,
+    };
 
-    let toml = toml::to_string(&s)?;
+    if !api_key {
+        // Default: use API token.
+        message::info("Looking to use a Global API Key and email instead? Run \"wrangler config --api-key\". (Not Recommended)");
+        println!("Enter API token: ");
+        let mut api_token_str: String = read!("{}\n");
+        api_token_str.truncate(api_token_str.trim_end().len());
+        if !api_token_str.is_empty() {
+            user.api_token = Some(api_token_str);
+        }
+    } else {
+        message::warn("We don't recommend using your Global API Key! Please consider using an API Token instead. https://support.cloudflare.com/hc/en-us/articles/200167836-Managing-API-Tokens-and-Keys");
+        println!("Enter email: ");
+        let mut email_str: String = read!("{}\n");
+        email_str.truncate(email_str.trim_end().len());
+        if !email_str.is_empty() {
+            user.email = Some(email_str);
+        }
+
+        println!("Enter global API key: ");
+        let mut api_key_str: String = read!("{}\n");
+        api_key_str.truncate(api_key_str.trim_end().len());
+        if !api_key_str.is_empty() {
+            user.api_key = Some(api_key_str);
+        }
```

Comment:
shouldn't this else be moved to the level of `if !api_key {} else {`, not if the API token string is empty? If the API token string is empty, I'd expect an error to be thrown (might as well check for length as well), but I wouldn't expect it to just continue to entering api token

Reply:
ah, so our baseline was that an empty string would not be serialized and would trigger the default API authentication failure message. This is preserving the preexisting behavior, but we can revisit this and think about adding additional error messages. We can do this after the RC is cut.